src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,28 @@
 package mate.academy;
 
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final CopyOnWriteArrayList<EventListener> events = new CopyOnWriteArrayList<>();
+    private final ExecutorService executor = Executors.newCachedThreadPool();
 
+    public void registerListener(EventListener listener) {
+        events.addIfAbsent(listener);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        events.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        for (EventListener listener : events) {
+            executor.submit(() -> listener.onEvent(event));
+        }
     }
 
     public void shutdown() {
-
+        executor.shutdown();
```

Comment:
The `shutdown` method calls `executor.shutdown()`, which initiates an orderly shutdown in which previously submitted tasks are executed, but no new tasks will be accepted. Consider using `executor.shutdownNow()` if you want to attempt to stop all actively executing tasks and halt the processing of waiting tasks. Additionally, you might want to check if the executor has terminated using `executor.awaitTermination()` to ensure all tasks have completed before proceeding.